Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js

```diff
@@ -546,12 +546,16 @@ window.__firefox__.includeOnce("LoginsHelper", function() {
 
           if (!disabledOrReadOnly && !userEnteredDifferentCase && userNameDiffers) {
             usernameField.value = selectedLogin.username;
+            // Fire 'change' event to notify client-side validation on this field.
+            usernameField.dispatchEvent(new Event("change"));
             dispatchKeyboardEvent(usernameField, "keydown", KEYCODE_ARROW_DOWN);
             dispatchKeyboardEvent(usernameField, "keyup", KEYCODE_ARROW_DOWN);
```

Comment:
To better match other browsers, regular typing, and desktop Firefox, the `change` event should be dispatched after keydown and keyup events but only if the field isn't already focused (webkit should automatically dispatch the change event upon blurring the field in that case).

Ideally an `input` event would be dispatched between `keydown` and `keyup` with `inputType: 'insertReplacementText'` but it seems like Webkit on iOS doesn't support the InputEvent constructor init arguments yet: https://bugs.webkit.org/show_bug.cgi?id=170416

Reply:
Hi, thanks for your review and sorry for late reply, I was on PTO.

I moved the code below keyboard dispatch events and guarded it with `document.activeElement` checks